src/fetch/period_slippage.py

```diff
@@ -19,12 +22,32 @@ def slippage_query(dune: DuneAnalytics) -> str:
     )
 
 
+@dataclass
+class SolverSlippage:
+    """Total amount reimbursed for accounting period"""
+    solver_address: Address
+    solver_name: str
+    # ETH amount (in WEI) to be deducted from Solver reimbursement
+    amount_wei: int
+
+    @classmethod
+    def from_dict(cls, obj: dict) -> SolverSlippage:
+        """Converts Dune data dict to object with types"""
+        return cls(
+            solver_address=Address(obj['solver_address']),
+            solver_name=obj['solver_name'],
+            # TODO - Reverse the signature of the value
+            amount_wei=int(obj['eth_slippage_wei'])
+        )
+
+
 def get_period_slippage(
         dune: DuneAnalytics,
         period_start: datetime,
         period_end: datetime,
-) -> list:
-    return dune.fetch(
+        allow_positive: bool = True
```

Comment:
Generally I agree with your suggestion, but I think it's important to remember that this means we would then be executing an expense query twice that returns identical results. Perhaps I can work this in a way so that the return type of the function returns two lists (one negative and one positive).

Reply:
Fixed. We now have a `SolverSlippages` class storing the negatives and positives separately. I hope you like it!